pkgs/pkgs-lib/formats/libconfig/default.nix

```diff
@@ -0,0 +1,68 @@
+{ pkgs, lib, ... }:
+rec {
+  # Implementation notes:
+  #   Libconfig spec: https://hyperrealm.github.io/libconfig/libconfig_manual.html
+  #
+  #   Since libconfig does not allow setting names to start with an underscore,
+  #   this is used as a prefix for both special types and include directives.
+  #
+  #   The difference between 32bit and 64bit values became optional in libconfig
+  #   1.5, so we assume 64bit values for all numbers.
+  generator = pkgs.rustPlatform.buildRustPackage {
+    name = "libconfig-generator";
+    version = "0.1.0";
+    src = ./.;
+
+    cargoLock.lockFile = ./Cargo.lock;
+  };
+
+  format = {}: {
+    type = with lib.types;
+      let
+        valueType = (oneOf [
+          bool
+          int
+          float
+          str
+          path
+          (attrsOf valueType)
+          (listOf valueType)
+        ]) // {
+          description = "libconfig value";
+        };
+      in
+        attrsOf valueType;
+
+    lib = {
+      mkHex = value: {
+        _type = "hex";
+        inherit value;
+      };
+      mkOctal = value: {
+        _type = "octal";
+        inherit value;
+      };
+      mkFloat = value: {
+        _type = "float";
+        inherit value;
+      };
+      mkArray = value: {
+        _type = "array";
+        inherit value;
+      };
+      mkList = value: {
+        _type = "list";
+        inherit value;
+      };
+    };
+
+    generate = name: value:
+      pkgs.runCommandLocal name
+        {
+          json = builtins.toJSON value;
+          passAsFile = [ "json" ];
+        } ''
+        < $jsonPath ${generator}/bin/libconfig-generator > "$out"
+      '';
+  };
```

Comment:
`validator.c` used to be ran on every generation to check it against the C `libconfig` source of truth in #208747 — now it seems it's been confined to a test. Why?

Reply:
I think using rust requires us to do so much parsing and validation already that we might just do it all the way. The idea was to use this test to check that our validator agrees with the library for every type (assuming I've remembered to add them all). Doing two runs of what *should* be the same validation logic on every config sounds wasteful.

Reply:
That seems to be based on the assumption of a large input, which I hadn't considered. I think correctness still comes first though.

Reply:
I guess I can add it to the `checkPhase`, so people can override the `dontCheck` flag if they want to. It feels overkill though.